src/libs/ajax/DataRepo.ts

```diff
@@ -143,6 +145,8 @@ export interface DataRepoContract {
     createSnapshotRequest(request: DatasetAccessRequest): Promise<DatasetAccessRequestApi>;
     getCounts(request: DatasetParticipantCountRequest): Promise<DatasetParticipantCountResponse>;
     getConcepts(parent: SnapshotBuilderConcept): Promise<GetConceptsResponse>;
+    // Search returns a list of matching concepts with a domain sorted by participant count. The result is truncated to N concepts.
+    searchConcepts(domain: SnapshotBuilderConcept, text: string): Promise<SearchConceptsResponse>;
```

Comment:
I'm not sure what you mean. Once a concept is in the cart, it will show up with a red minus sign next to it so a user can remove it from the cart. This state is preserved across searches, if you select an item, do a filtered search, the item will retain its original state with respect to the cart.

Maybe you're suggesting a change to the behavior of search relative to the current cart contents?

Reply:
Sorry for the confusion, I'm suggesting a change to the behavior of search to the current cart contents. Basically, a way of filtering out searched concepts that are already in the cart, that way the same item cannot be added twice. This video may add more clarification! 

https://github.com/DataBiosphere/terra-ui/assets/63474660/7c94cf8a-fc80-42f6-a0a1-db36af4d55d0